File: app/src/main/java/com/example/sarahn/waterreminderapp/classes/MyApplication.java
package com.example.sarahn.waterreminderapp.classes;

import android.app.Application;
import android.content.Context;

/**
 * Created by SarahN on 6/28/2017.
 */
public class MyApplication extends Application {

    private static Context mContext;

    @Override
    public void onCreate() {
        super.onCreate();
        mContext = getApplicationContext();
    }

    public static Context getContext() {
        return mContext;
    }

    //getting things on main thread and application context managment of code
}

File: app/src/main/java/com/example/sarahn/waterreminderapp/classes/MyApplication.kt
package com.example.sarahn.waterreminderapp.classes

import android.app.Application
import android.content.Context

/**
 * Created by SarahN on 6/28/2017.
 */
class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        context = applicationContext
    }

    companion object {
        //getting things on main thread and application context managment of code
        var context: Context? = null
            private set
    }
}